Unit-Testing-Framework/test/specs/calculator/add.spec.js

```diff
@@ -0,0 +1,57 @@
+const Calculator = require('../../../app/calculator.js');
+const { expect } = require('chai');
+
+describe('is addition positive case', function () {
+    let numbers;
+    beforeEach(function () {
+        numbers = new Calculator();
+    });
+    afterEach(function () {
+        numbers = null;
+    });
+    it('should return the sum when adding two integers', function () {
+        expect(numbers.add(2, 8)).to.be.equal(10);
+    });
+    it('should return the sum of two fractional numbers ', function () {
+        expect(numbers.add(0.5, 0.6)).to.be.equal(1.1);
+    });
+    it('should return the sum when one argument is an integer and the other is fractional ', function () {
+        expect(numbers.add(10, 2.5)).to.be.equal(12.5);
+    });
+    it('should return the sum, which will be negative if the two arguments are negative numbers', function () {
+        expect(numbers.add(-5, -5)).to.be.equal(-10);
+    });
```

Comment:
These tests can be simplified with dataproviders.

https://gist.github.com/burkhardr/54162f667ff756540574082f61350b38

Reply:
done with mocha data provider